pkg/registry/resource/resourceclaim/storage/storage.go

```diff
@@ -38,19 +41,20 @@ type REST struct {
 }
 
 // NewREST returns a RESTStorage object that will work against ResourceClaims.
-func NewREST(optsGetter generic.RESTOptionsGetter) (*REST, *StatusREST, error) {
+func NewREST(optsGetter generic.RESTOptionsGetter, nsClient v1.NamespaceInterface) (*REST, *StatusREST, error) {
+	strategy := resourceclaim.NewStrategy(legacyscheme.Scheme, names.SimpleNameGenerator, nsClient)
```

Comment:
if nsClient is nil, return an error (it'll panic later if an adminAccess object is attempted to be created, we want to detect/error at construction time rather than later at runtime)